yt_dlp/extractor/vidyard.py

```diff
@@ -0,0 +1,156 @@
+from .common import InfoExtractor
+from ..utils import int_or_none, traverse_obj
+
+
+class VidyardBaseInfoExtractor(InfoExtractor):
+
+    def _get_formats_and_subtitle(self, video_source, video_id):
+        video_source = video_source or {}
+        formats, subtitles = [], {}
+        for key, value in video_source.items():
+            if key == 'hls':
+                for video_hls in value:
+                    fmts, subs = self._extract_m3u8_formats_and_subtitles(video_hls.get('url'), video_id, headers={
+                        'referer': 'https://play.vidyard.com/'
+                    })
+                    formats.extend(fmts)
+                    self._merge_subtitles(subs, target=subtitles)
+            else:
+                formats.extend({
+                    'url': video_mp4.get('url'),
+                    'ext': 'mp4'
+                } for video_mp4 in value)
+
+        return formats, subtitles
+
+    def _get_direct_subtitle(self, caption_json):
+        subs = {}
+        for caption in caption_json:
+            subs.setdefault(caption.get('language') or 'und', []).append({
+                'url': caption.get('vttUrl'),
+                'name': caption.get('name')
+            })
+
+        return subs
+
+    def _webpage_url(self, url, video_id):
+        return url
+
+    def _real_extract(self, url):
+        video_id = self._match_valid_url(url).group('id')
+        webpage = self._download_webpage(self._webpage_url(url, video_id), video_id)
```

Comment:
This is not being used for mandatory data, is it?
```suggestion
        webpage = self._download_webpage(self._webpage_url(url, video_id), video_id, default='')
```
If that change is accepted the line could be moved down until needed.

Reply:
As of https://github.com/yt-dlp/yt-dlp/pull/10155/commits/0b7a3592721742170378d561fbf0e41559d0fd16 webpage download now only happens if title/description cannot be retrieved from JSON